apisix/init.lua

```diff
@@ -512,6 +512,22 @@ function _M.http_access_phase()
         return pubsub_kafka.access(api_ctx)
     end
 
+    if api_ctx.matched_upstream and api_ctx.matched_upstream.tls_id then
```

Comment:
Could we move this just after the upstream is found:
https://github.com/apache/apisix/blob/975038ea70dc47e0613c42b3eabb8dcc730216a4/apisix/init.lua#L497